iceoryx_hoofs/include/iceoryx_hoofs/cxx/requires.hpp

```diff
@@ -33,8 +33,12 @@ void Require(
 // see:
 // https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Ri-expects
 /// @NOLINTJUSTIFICATION macro required to capture file, line, function origin of call implicitly
-/// @NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+/// @NOLINTBEGIN(cppcoreguidelines-macro-usage)
+/// @NOLINTJUSTIFICATION array decay: needed for source code location, safely wrapped in macro
+/// @NOLINTBEGIN(hicpp-no-array-decay, cppcoreguidelines-pro-bounds-array-to-pointer-decay)
```

Comment:
I wasn't able to combine all three warnings since this introduces a line break in `NOLINTBEGIN` but not in `NOLINTEND` and with this, clang-tidy seems to have problems to match both.

Reply:
This is not bad, since `Expects` will be replaced anyway. I saw this myself as well (should probably add `NOLINTJUSTIFICATION` line in my PR as well. 

I really think this is all way to verbose in general (not only this particular case, the whole result for sometimes trivial suppressions) :-(